nodebuilder/blobstream/service.go

```diff
@@ -0,0 +1,534 @@
+// It takes as argument the height of the block containing the blob of data, its
+// namespace and its share commitment.
+func (s *Service) ProveCommitment(ctx context.Context, height uint64, namespace share.Namespace, shareCommitment []byte) (*ResultCommitmentProof, error) {
+	log.Debugw("proving share commitment", "height", height, "commitment", shareCommitment, "namespace", namespace)
+	if height == 0 {
+		return nil, fmt.Errorf("height cannot be equal to 0")
+	}
+
+	// get the blob to compute the subtree roots
+	log.Debugw("getting the blob", "height", height, "commitment", shareCommitment, "namespace", namespace)
+	blb, err := s.blobServ.Get(ctx, height, namespace, shareCommitment)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugw("converting the blob to shares", "height", height, "commitment", shareCommitment, "namespace", namespace)
+	blobShares, err := blob.BlobsToShares(blb)
+	if err != nil {
+		return nil, err
+	}
+	if len(blobShares) == 0 {
+		// TODO we return the share commitment as hex or some other format?
+		return nil, fmt.Errorf("the blob shares for commitment %s are empty", hex.EncodeToString(shareCommitment))
+	}
+
+	// get the extended header
+	log.Debugw("getting the extended header", "height", height)
+	extendedHeader, err := s.headerServ.GetByHeight(ctx, height)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugw("getting eds", "height", height)
+	eds, err := s.shareServ.GetEDS(ctx, extendedHeader)
+	if err != nil {
+		return nil, err
+	}
+
+	// find the blob shares in the EDS
+	blobSharesStartIndex := -1
+	for index, share := range eds.FlattenedODS() {
+		if bytes2.Equal(share, blobShares[0]) {
+			blobSharesStartIndex = index
+		}
+	}
+	if blobSharesStartIndex < 0 {
+		return nil, fmt.Errorf("couldn't find the blob shares in the ODS")
+	}
+
+	nID, err := appns.From(namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugw("generating the blob share proof for commitment", "commitment", shareCommitment, "start_share", blobSharesStartIndex, "end_share", blobSharesStartIndex+len(blobShares), "height", height)
+	sharesProof, err := pkgproof.NewShareInclusionProofFromEDS(eds, nID, shares.NewRange(blobSharesStartIndex, blobSharesStartIndex+len(blobShares)))
+	if err != nil {
+		return nil, err
+	}
+
+	// convert the shares to row root proofs to nmt proofs
+	var nmtProofs []*nmt.Proof
+	for _, proof := range sharesProof.ShareProofs {
+		nmtProof := nmt.NewInclusionProof(int(proof.Start),
+			int(proof.End),
+			proof.Nodes,
+			true)
+		nmtProofs = append(
+			nmtProofs,
+			&nmtProof,
+		)
+	}
+
+	// compute the subtree roots of the blob shares
+	log.Debugw("computing the subtree roots", "height", height, "commitment", shareCommitment, "namespace", namespace)
+	var subtreeRoots [][]byte
+	var dataCursor int
+	for _, proof := range nmtProofs {
+		// TODO: do we want directly use the default subtree root threshold or want to allow specifying which version to use?
```

Comment:
[note to reviewers]
do we want directly use the default subtree root threshold or want to allow specifying which version to use?